File: go/pkg/config/node.go
/** Go types for node configuration.
 *
 * @author Steffen Vogel <svogel2@eonerc.rwth-aachen.de>
 * @copyright 2014-2022, Institute for Automation of Complex Power Systems, EONERC
 * @license Apache 2.0
 *********************************************************************************/

package config

type Node struct {
	Name string `json:"name"`
	Type string `json:"type"`
}

type NodeDir struct {
}

type NodeLoopbackIn struct {
	NodeDir

	Signals []Signal      `json:"signals"`
	Hooks   []interface{} `json:"hooks"`
}

type LoopbackNode struct {
	Node

	In NodeLoopbackIn `json:"in"`
}

File: lib/api/requests/node.cpp
/* Node API Request.
 *
 * Author: Steffen Vogel <post@steffenvogel.de>
 * SPDX-FileCopyrightText: 2014-2023 Institute for Automation of Complex Power Systems, RWTH Aachen University
 * SPDX-License-Identifier: Apache-2.0
 */

#include <villas/api/requests/node.hpp>

using namespace villas::node::api;

void NodeRequest::prepare() {
  int ret;

  auto &nodes = session->getSuperNode()->getNodes();

  uuid_t uuid;
  ret = uuid_parse(matches[1].c_str(), uuid);
  if (ret) {
    node = nodes.lookup(matches[1]);
    if (!node)
      throw BadRequest("Unknown node", "{ s: s }", "node", matches[1].c_str());
  } else {
    node = nodes.lookup(uuid);
    if (!node)
      throw BadRequest("No node found with with matching UUID", "{ s: s }",
                       "uuid", matches[1].c_str());
  }
}
